src/connector/codec/src/decoder/avro/mod.rs

```diff
@@ -107,6 +108,51 @@ impl<'a> AvroParseOptions<'a> {
 
         let v: ScalarImpl = match (type_expected, value) {
             (_, Value::Null) => return Ok(DatumCow::NULL),
+            // ---- Union (with >=2 non null variants) -----
+            (DataType::Struct(struct_type_info), Value::Union(variant, v)) => match self.schema {
+                Some(Schema::Union(u)) => {
+                    if let Some(inner) = get_nullable_union_inner(u) {
+                        // nullable Union ([null, record])
```

Comment:
Top comment is a little bit misleading: it covers both `Union (with >=2 non null variants)`, and `nullable Union ([null, record])`